opentelemetry-sdk/src/opentelemetry/sdk/_metrics/export/__init__.py

```diff
@@ -77,3 +90,116 @@ def export(self, metrics: Sequence[Metric]) -> MetricExportResult:
+    that collects metrics based on a user-configurable time interval, and passes the
+    metrics to the configured exporter.
+    """
+
+    def __init__(
+        self,
+        exporter: MetricExporter,
+        export_interval_millis: Optional[float] = None,
+        export_timeout_millis: Optional[float] = None,
+    ) -> None:
+        self._exporter = exporter
+        if export_interval_millis is None:
+            try:
+                export_interval_millis = float(
+                    environ.get("OTEL_METRIC_EXPORT_INTERVAL", 60000)
+                )
+            except ValueError:
+                export_interval_millis = 60000
+        if export_timeout_millis is None:
+            try:
+                export_timeout_millis = float(
+                    environ.get("OTEL_METRIC_EXPORT_TIMEOUT", 30000)
+                )
+            except ValueError:
+                export_timeout_millis = 30000
+        self._export_interval_millis = export_interval_millis
+        self._export_timeout_millis = export_timeout_millis
+        self._meter_provider = None  # type: MeterProvider
+        self._shutdown = False
+        self._daemon_thread = threading.Thread(
+            target=self.collect, daemon=True
+        )
+        self._condition = threading.Condition()
+        self._flush_event = None  # type: Optional[threading.Event]
+        self._lock = threading.Lock()
+        self._daemon_thread.start()
+        if hasattr(os, "register_at_fork"):
+            os.register_at_fork(
+                after_in_child=self._at_fork_reinit
+            )  # pylint: disable=protected-access
+
+    def _at_fork_reinit(self):
+        self._condition = threading.Condition()
+        self._lock = threading.Lock()
+        self._daemon_thread = threading.Thread(
+            target=self.collect, daemon=True
+        )
+        self._daemon_thread.start()
+
+    def _register_meter_provider(self, provider: MeterProvider) -> None:
+        self._meter_provider = provider
+
+    def collect(self) -> None:
+        while not self._shutdown:
+            with self._condition:
+                # might have received flush/shutdown request
+                if not self._flush_pending and not self._shutdown:
+                    self._condition.wait(self._export_interval_millis / 1e3)
+                # any pending flush request would be fulfilled later
+                if self._shutdown:
+                    break
+            # export and set flush event if exists
+            self._export()
+        # one last flush below before shutting down completely
+        self._export()
+
+    @property
+    def _flush_pending(self) -> bool:
+        return self._flush_event is not None
+
+    def _export(self) -> None:
+        token = attach(set_value(_SUPPRESS_INSTRUMENTATION_KEY, True))
+        try:
+            metrics = []
+            for reader in self._meter_provider._metric_readers:
+                metrics.extend(reader.collect())
```

Comment:
A given MetricReader should only be collecting for itself, this shouldn't be necessary

Reply:
I am little confused. Will there be periodic metric reader for each metrics reader passed to provider?